lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_async_setstate/avoid_async_setstate_rule.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../../../../../utils/node_utils.dart';
+import '../../../lint_utils.dart';
+import '../../../models/internal_resolved_unit_result.dart';
+import '../../../models/issue.dart';
+import '../../../models/severity.dart';
+import '../../models/flutter_rule.dart';
+import '../../rule_utils.dart';
+
+part 'visitor.dart';
+part 'helpers.dart';
+
+class AvoidAsyncSetStateRule extends FlutterRule {
```

Comment:
Thinking more about the name and looks like it can confuse people like it bans `setState(async () {});`. Maybe you have other ideas how to name it?

Reply:
We could follow the precedence here and call it `use-setstate-synchronously`, what do you think?

Reply:
Yeah, IMO, this one is better 👍